### ISSUE TITLE
Update Fails When Admin Differs From User Account

### ISSUE THREAD

**Issue by Nosamdaman** [git-for-windows/git#3677]
- [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.35.1.windows.1
cpu: x86_64
built from commit: 8084c8c201ddb28eaff22bdd9c8344339eddab73
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19044.1503]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Powershell 7

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git update-git-for-windows
```
 - What did you expect to occur after running these commands?

I expected the update to complete. Due to a company policy, we actually have two user accounts, one for daily use and one with administrative privileges. I started the update in my usual user account, but when prompted for admin, I gave the credentials for my admin account.

 - What actually happened instead?

The update started just fine, but after uninstalling the current version of git, it failed to update to the latest version.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** insert URL here **

**Comment by dscho** [comment]
> it failed to update to the latest version.

Can you provide more details about the symptoms?

**Comment by Nosamdaman** [comment]
> Can you provide more details about the symptoms?
Updated with more information

**Comment by dscho** [comment]
Hmm. The `dev` directory is supposed to be removed by the uninstaller: https://github.com/git-for-windows/build-extra/blob/e97e9181c01546f930be9123414ca2cf3c9b91de/installer/install.iss#L138

Are there any files or directories left in `C:\Program Files\Git` after the uninstaller ran (please list them recursively, if there are)? Also, what was the previous Git for Windows version that was uninstalled (which might not have included https://github.com/git-for-windows/build-extra/commit/b0c40ad8e7705b0e8aaaea27dc9b1158458dd7bf)?

**Comment by Nosamdaman** [comment]
Updated

**Comment by dscho** [comment]
Strange that the `dev` directory does not exist. Can you create it (as non-admin)?

**Comment by Nosamdaman** [comment]
> Strange that the `dev` directory does not exist. Can you create it (as non-admin)?

No I cannot

**Comment by dscho** [comment]
> No I cannot

Even though it does not exist? That's strange.

The only times I encountered anything similar to this was when some files or directories were still in use, and the Git for Windows installer suggested to reboot before continuing (but of course I did not reboot, everything worked fine, right up until the time I wanted to upgrade Git for Windows _again_).

**Comment by Nosamdaman** [comment]
I agree this is strange, my best guess is that is has to do with switching users to get admin privileges. I've seen similar issues/quirks with other programs. It'd be great is someone could try to reproduce this issue.

**Comment by vbaderks** [comment]
On my personal dev system I am also using 2 accounts: a normal one and an admin account. I have also seen this issue in the past when upgrading to a newer version of git. My typical workaround is to run the git installer from a cmd console which is already running under the admin account. I have not analyzed the precise location of the problem in the installer.

**Comment by a563151** [comment]
I also encounter this issue with every update.
The company I work for uses [Admin By Request](https://www.adminbyrequest.com/)" to grant privileges when doing "admin" stuff.

### How it goes for me:

I am notified by popup that there is a new version of Git for Windows.
I chose "Yes" to install the update.
For the uninstall part, a popup requesting admin privileges comes up. After doing the "request dance" in Admin By Request, I can now use the **current user** credentials as admin credentials, something that would have failed without the "request dance". Uninstall is successful.
The install part is next, but I get no prompt for admin credentials, and the installer fails as described above. I am now left with no Git for Windows on my machine.

### How I resolve this:

After the botched uninstall + install, I use "Run as administrator" on the installer, I do the "request dance", provide the credentials for the current user that now has admin privileges. Install is successful. (but I had to go through ALL the configuration options again)

**Comment by Nosamdaman** [comment]
Yeah my workaround is the same as @vbaderks, opening an admin terminal first then running the update works for me, thought it would be nice to be able to just run `winget upgrade` and have it work.

**Comment by mfriedrich74** [comment]
Hi,

Just a hunch, this might have to do with InnoSetup running some things
elevated and some non elevated as current user. It may fail on anything
that operates using current non elevated user.
When using a CMD already elevated prompt, it will run all thing elevated
(current user resolves to the elevated account also).
This may be able to be fixed by specifying scripts in run and uninstall
sections to use the elevated user (swap runasoriginaluser for
runascurrentuser).
Not sure which is default.
If it's not from scripts, then other sections may have similar switches?

If the bug is in the uninstaller though, it might be hard to fix.

Best regards, Mike
